services/galley/src/Galley/API/MLS/Message.hs

```diff
@@ -796,10 +796,29 @@ processInternalCommit qusr senderClient con lconv cm epoch groupId action sender
     postponedKeyPackageRefUpdate <-
       if epoch == Epoch 0
         then do
-          -- this is a newly created conversation, and it should contain exactly one
-          -- client (the creator)
-          case (self, cmAssocs cm) of
-            (Left lm, [(qu, (creatorClient, _))])
+          let cType = cnvmType . convMetadata . tUnqualified $ lconv
+          case (self, cType, cmAssocs cm) of
+            (Left _, SelfConv, []) -> do
+              creatorClient <-
+                note (mlsProtocolError "Missing the sender client") senderClient
```

Comment:
It's probably better to make this a 403. It might be good to include more context here in the error message. Something that explains that the client has to refresh their access token and provide their client ID.

Reply:
Thanks! Addressed it in commit 341aeddde (to be pushed soon).